File: insfrastructure/usecases/withdraw_from_account_usecase.py
from insfrastructure.entities.person_entity import PersonEntity
from insfrastructure.enums.account_types import AccountType
from insfrastructure.repositories.bank_account_repository import BankAccountRepository


class WithdrawFromAccountUseCase:
    def __init__(self, bank_account_repo: BankAccountRepository):
        self.bank_account_repo = bank_account_repo

    def execute(self, account_number: int, amount: float) -> bool:
        return self.bank_account_repo.withdraw(account_number, amount)

File: typescript-impl/src/infrastructure/usecases/withdraw_from_account_usecase.ts
import { BankAccountRepository } from '../repositories/bank_account_repository';

export class WithdrawFromAccountUseCase {
    constructor(private bank_account_repo: BankAccountRepository) { }

    execute(account_number: number, amount: number): boolean {
        return this.bank_account_repo.withdraw(account_number, amount);
    }
}
